libs/api/rpc_client.h

```diff
@@ -14,3 +14,6 @@ clean_up_rpc_binding(void);
 
 int
 ebpf_rpc_verify_program(ebpf_program_verify_info* info, const char** logs, uint32_t* logs_size);
+
+int
```

Comment:
Shouldn't the return type be an enum like ebpf_result_t?  Or is that part of another github issue?  if the latter then can you add a // comment with a TODO that references the github issue #?

Reply:
Currently the ebpf_api_* APIs return uint32_t instead of ebpf_result_t. I can fix this when I start changing the APIs.

Reply:
It would be good if we can stop adding more debt and at least make any new additions use ebpf_result_t instead of adding more code we know is "wrong" (I gave the same feedback on Shankar's PR btw).

Reply:
Agree. One question regarding guidance: Is it OK for internal functions use uint32_t and only the public APIs (and RPC APIs) return ebpf_result_t? 

Example, in this function (ebpf_rpc_verify_program()), this function returns RPC exception code if there is an RPC exception. If we change this function to return ebpf_result_t, rpc exception code will get absorbed.

Reply:
uint32_t is a really bad type to use for things with discrete values that could be an enum, since they make debugging and maintainability far more difficult.  So one should always use an enum type instead of uint32_t even for internal functions, whenever possible.

Reply:
changed these functions to return ebpf_result_t